app/controllers/question_controller.rb

```diff
@@ -2,8 +2,8 @@ class QuestionsController < ApplicationController
 	include ActionView::Helpers::TextHelper
 
 	before_action :authenticate_user!, except: [:index, :show]
-	before_action :set_question, only: [:show, :edit, :update, :destroy]
-
+	before_action :question # only: [:show, :edit, :update, :destroy]
+	before_action :user
```

Comment:
Why do we need a user id for viewing a question or the index? I don't think I understand the def user method. Probably just not thinking about this straight, but what does this do?